### PR TITLE
[Feature][Doc] The contribute document need be update

### PR DESCRIPTION
**Pull request by Mr-LiuXu**
..was encountered
https://github.com/apache/incubator-seatunnel/issues/2541

## Purpose of this pull request

The contribute document in `seatunnel/docs/en/contribution/contribute-plugin.md` are out of date. We should add `connector-v2` contribute guide in it. And fix `plugin-mapping.properties` path in doc. And add any information which can help new developer start.

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Review by CalvinKirs — COMMENTED** [review]
I know that we still have some public parameters, such as schema, and some common basic modules, such as file-base, format, etc., can we add these?

File: docs/en/contribution/contribute-plugin.md [hunk 1]
```diff
@@ -140,3 +140,109 @@ start a Clickhouse database in your test. You can use `GenericContainer` to star
...
+
+We realize the conversion between SeaTunnel API and Engine API by adapting the interfaces of different engines, so as to achieve the effect of translation, and let our SeaTunnel Connector support the operation of multiple different engines. The corresponding code address, ``seatunnel-translation``, this module has the corresponding translation layer implementation. If you are interested, you can view the code and help us improve the current code.
+
+## API introduction
+
+The API design of the current version of SeaTunnel draws on the design concept of Flink.
+
+### Source
+
+#### SeaTunnelSource.java
```
Comment (hailin0) [inline docs/en/contribution/contribute-plugin.md]: link to connector-v2-readme?

https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-connectors-v2

Comment (Mr-LiuXu) [reply docs/en/contribution/contribute-plugin.md]: Yes, it has been modified to link to connector-v2-readme

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**